x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/index.tsx

```diff
@@ -43,138 +37,107 @@ const TIMELINE_MARGINS = {
   bottom: 0
 };
 
+const toggleFlyout = ({
+  item,
+  location
+}: {
+  item?: IWaterfallItem;
+  location: Location;
+}) => {
+  history.replace({
+    ...location,
+    search: fromQuery({
+      ...toQuery(location.search),
+      ...{
+        flyoutDetailTab: undefined,
+        waterfallItemId: item ? String(item.id) : undefined
+      }
+    })
+  });
+};
+
+const WaterfallItemsContainer = styled.div<{
+  paddingTop: number;
+}>`
+  padding-top: ${props => px(props.paddingTop)};
+`;
+
 interface Props {
-  agentMarks: AgentMark[];
-  urlParams: IUrlParams;
+  waterfallItemId?: string;
   waterfall: IWaterfall;
   location: Location;
-  serviceColors: IServiceColors;
   exceedsMax: boolean;
 }
 
-export class Waterfall extends Component<Props> {
-  public onOpenFlyout = (item: IWaterfallItem) => {
-    this.setQueryParams({
-      flyoutDetailTab: undefined,
-      waterfallItemId: String(item.id)
-    });
-  };
+export const Waterfall: React.FC<Props> = ({
+  waterfall,
+  exceedsMax,
+  waterfallItemId,
+  location
+}) => {
+  const itemContainerHeight = 58; // TODO: This is a nasty way to calculate the height of the svg element. A better approach should be found
+  const waterfallHeight = itemContainerHeight * waterfall.items.length;
 
-  public onCloseFlyout = () => {
-    this.setQueryParams({
-      flyoutDetailTab: undefined,
-      waterfallItemId: undefined
-    });
-  };
+  const { serviceColors, duration } = waterfall;
 
-  public renderWaterfallItem = (item: IWaterfallItem) => {
-    const { serviceColors, waterfall, urlParams }: Props = this.props;
+  const agentMarkers = getAgentMarks(waterfall.entryTransaction);
+  const errorMarkers = waterfall.items.filter(item => item.docType === 'error');
```

Comment:
Just like we have `getAgentMarks` I think we should have `getErrorMarks` which returns a `Mark` type (or something like it). Idea being that `agentMarkers` and `errorMarkers` have the same interface.

Btw. why agentMarkers instead of `agentMarks` ?